common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java

```diff
@@ -375,6 +376,16 @@ protected void renewAccessToken(@NonNull final SilentTokenCommandParameters para
                                     + isRemoved
                     );
                 }
+
+                /*
+                    Intended to cover the AAD outage scenario for the refresh_in logic.
+                    Should return existing AT without refreshing it.
+                    This way caller will know whether to refresh based on this exception.
+                 */
+                if(SERVICE_NOT_AVAILABLE.equals(errorCode)){
+                    throw new ServiceException(SERVICE_NOT_AVAILABLE, "AAD is not available.", tokenResult.getErrorResponse().getStatusCode(), null);
+                }
```

Comment:
What was the existing behavior? I would think that ServiceException was already being thrown somewhere

Reply:
Currently the only relevant place, to this PR, from which we are throwing a ServiceException, is from BaseController.isRequestAuthorityRealmSameAsATRealm(), which is called within the LocalMsalController.acquireTokenSilent() - line ~353, as part of a larger set of conditions for determining if we need a new AT. This ServiceException is thrown for isRequestAuthorityRealmSameAsATRealm() because it talks to server side, but I'm not sure why it's not thrown as a result of renewAccessToken, which also talks to server side.